src/fides/api/models/privacy_notice.py

```diff
@@ -332,123 +365,249 @@ def create(
+                    db,
+                    data={
+                        **history_data,
+                        **updated_translation_data,
+                        "translation_id": translation.id,
+                        "version": existing_version + 1.0,
+                    },
+                    check_name=False,
+                )
 
-        if updated:
-            history_data = create_historical_data_from_record(resource)
-            history_data["privacy_notice_id"] = resource.id
-            PrivacyNoticeHistory.create(db, data=history_data, check_name=False)
+        # Removes any translations not supplied in the request from the Notice
+        notice_translations: List[NoticeTranslation] = self.translations
+        translations_to_remove: Set[SupportedLanguage] = set(  # type: ignore[assignment]
+            translation.language for translation in notice_translations
+        ).difference(
+            set(translation.get("language") for translation in request_translations)
+        )
 
-        return resource  # type: ignore[return-value]
+        db.query(NoticeTranslation).filter(
+            NoticeTranslation.language.in_(translations_to_remove),
+            NoticeTranslation.privacy_notice_id == self.id,
+        ).delete()
+        db.commit()
 
+        return self  # type: ignore[return-value]
 
-PRIVACY_NOTICE_TYPE = Union[PrivacyNotice, PrivacyNoticeTemplate]
+    def dry_update(self, *, data: dict[str, Any]) -> PrivacyNotice:
+        """
+        A utility method to get an updated object without saving it to the db.
 
+        This is used to see what an object update would look like, in memory,
+        without actually persisting the update to the db
+        """
+        # Update our attributes with values in data
+        cloned_attributes = self.__dict__.copy()
+        for key, val in data.items():
+            cloned_attributes[key] = val
 
-def check_conflicting_notice_keys(
-    new_privacy_notices: Iterable[PRIVACY_NOTICE_TYPE],
-    existing_privacy_notices: Iterable[Union[PRIVACY_NOTICE_TYPE]],
-    ignore_disabled: bool = True,  # For PrivacyNoticeTemplates, set to False
-) -> None:
-    """
-    Checks to see if new notice keys will conflict with any existing notice keys for a specific region
-    """
-    # Map regions to existing notice key, notice name
-    notice_keys_by_region: Dict[
-        PrivacyNoticeRegion, List[Tuple[str, str]]
-    ] = defaultdict(list)
-    for privacy_notice in existing_privacy_notices:
-        if privacy_notice.disabled and ignore_disabled:
-            continue
-        for region in privacy_notice.regions:
-            notice_keys_by_region[PrivacyNoticeRegion(region)].append(
-                (privacy_notice.notice_key, privacy_notice.name)
+        # remove protected fields from the cloned dict
+        cloned_attributes.pop("_sa_instance_state", None)
+        cloned_attributes.pop("id", None)
+        cloned_attributes.pop("translations", [])
```

Comment:
_maybe_ some opportunity for extracting into a helper/utility that's shared with the `PrivacyExperienceConfig` dry updates? may not be worth it though!

Reply:
Yes I've been able to share my new `clone_object_attributes` and `dry_update_data` methods in this file 👍